src/Neo/Network/P2P/RemoteNode.cs

```diff
@@ -30,6 +30,12 @@ namespace Neo.Network.P2P
     /// </summary>
     public partial class RemoteNode : Connection
     {
+        public delegate void MessageSentHandler(NeoSystem system, Message message);
```

Comment:
Are these changes on Remote Node a general improvement or strictly connect to the Telemetry Plugin?

In any case I think the PR will need to be split.